portfolio/components/ui/Figures/CroppedAddressImage.tsx

```
@@ -0,0 +1,436 @@
+  const cropAspectRatio = box.w / box.h;
+  // Container height = width * (box.h / box.w) * (display.height / display.width)
+  //                  = width * (1 / cropAspectRatio) * (1 / displayAspectRatio)
+  const containerAspectRatio = cropAspectRatio * displayAspectRatio;
+
+  // Ensure aspect ratio is valid (prevent division by zero)
+  if (!isFinite(containerAspectRatio) || containerAspectRatio <= 0) {
+    console.error("Invalid aspect ratio:", containerAspectRatio, "box:", box, "display:", { displayWidth, displayHeight });
+  }
+
+  // Handle image load to get actual dimensions
+  const handleImageLoad = useCallback((e: React.SyntheticEvent<HTMLImageElement>) => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)